File: beagle-ui/src/main/java/br/com/zup/beagleui/engine/framework/layout/Horizontal.kt
package br.com.zup.beagleui.engine.framework.layout

import br.com.zup.beagleui.engine.framework.core.Flex
import br.com.zup.beagleui.engine.framework.core.Widget

data class Horizontal(
    val children: List<Widget>,
    val flex: Flex? = null,
    val reversed: Boolean? = null
) : Widget

File: Sources/BeagleUI/Sources/Widgets/Layout/Horizontal.swift
//
//  Horizontal.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

public struct Horizontal: Widget, HasAppearance {
    
    // MARK: - Public Properties
    
    public let appearance: Appearance?
    public let children: [Widget]
    public let reversed: Bool
    
    // MARK: - Initialization
    
    public init(
        appearance: Appearance? = nil,
        children: [Widget] = [],
        reversed: Bool = false
    ) {
        self.appearance = appearance
        self.children = children
        self.reversed = reversed
    }
    
    // MARK: - Configuration
    
    public func reversed(_ reversed: Bool = true) -> Horizontal {
        return Horizontal(children: children, reversed: reversed)
    }
    
}
